.github/workflows/publish_docker_images.yml

```diff
@@ -55,12 +57,131 @@ jobs:
 
       - name: Build and push
         uses: docker/build-push-action@v4
+        id: build
         with:
           context: .
-          push: true
+          platforms: linux/amd64
           build-args: |
             QW_COMMIT_DATE=${{ env.QW_COMMIT_DATE }}
             QW_COMMIT_HASH=${{ env.QW_COMMIT_HASH }}
             QW_COMMIT_TAGS=${{ env.QW_COMMIT_TAGS }}
-          tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
+          outputs: type=image,name=${{ env.REGISTRY_IMAGE }},push-by-digest=true,name-canonical=true,push=true
+
+      - name: Export digest
+        run: |
+          mkdir -p /tmp/digests
+          digest="${{ steps.build.outputs.digest }}"
+          touch "/tmp/digests/${digest#sha256:}"
+
+      - name: Upload digest
+        uses: actions/upload-artifact@v3
+        with:
+          name: digests
+          path: /tmp/digests/*
+          if-no-files-found: error
+          retention-days: 1
+
+  docker-arm:
```

Comment:
This job is almost a copy of the `docker` one. We can probably simplify the workflow using a matrix and a conditional step for QEMU.